torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -183,11 +183,14 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
             self._global_stride = param.stride()
             param_data = cast(DTensor, param)._local_tensor
         else:
-            if _mesh_resources.get_parent_mesh(self.mesh_info.mesh) is not None:
+            mesh = self.mesh_info.mesh
+            if (parent_mesh := _mesh_resources.get_parent_mesh(mesh)) is not None:
```

Comment:
hmmm what if user only apply TP to some submodules, and the other submodules are not DTensors but they are replicated (either manual replication or always replicated like torch.ones param). Would applying FSDP on top of this model fail?

Reply:
Today, it would fail. The question to me is what should be the sharded state dict look like, and we can go from there. If it is okay for us to mix 1D and 2D DTensors in the same sharded state dict, then we can simply relax this check instead of erroring.

Reply:
> If it is okay for us to mix 1D and 2D DTensors in the same sharded state dict

We do this today with FSDP1 and it works great :)

Reply:
Nice! Okay, we can probably relax this check then :)